tests/net/proptest.rs

```diff
@@ -0,0 +1,177 @@
+//! Proptest helpers and strategies
+//!
+//! This module houses strategies to generate (and reduce/expand) various `hbbft` and `net` related
+//! structures.
+
+use proptest::prelude::Rng;
+use proptest::strategy::{Strategy, ValueTree};
+use proptest::test_runner::{Reason, TestRunner};
+
+/// Node network topology.
+///
+/// A `NetworkTopology` describes the number of correct and faulty nodes in a network. It can also
+/// be checked, "averaged" (using the `halfway` function) and generated using `NetworkTopologyTree`.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct NetworkTopology {
```

Comment:
I'd reserve the term _topology_ for a representation of a node connectivity graph. Node connectivity might also play a role in tests, especially if we simulate p2p communication or network splits.

I suggest to use a more descriptive name that refers to correctness and faultiness of network nodes.
A few suggestions: `NetworkFaultiness` `NetworkFaultinessDegree`, `NetworkSizeWithFaults`.